kroxylicious-kms-provider-hashicorp-vault/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/config/TrustProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.hashicorp.vault.config;
+
+import javax.net.ssl.TrustManager;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+/**
+ * A TrustProvider is a source of trust anchors used to determine whether a certificate present by a peer is trusted.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.DEDUCTION)
+@JsonSubTypes({ @JsonSubTypes.Type(TrustStore.class), @JsonSubTypes.Type(InsecureTls.class) })
+public interface TrustProvider {
+
+    /**
+     * Return trust managers
+     * @return trust manager array, or null if this provider supplies no trust
+     */
+    TrustManager[] getTrustManagers();
```

Comment:
When do we want to support multiple `TrustManager`s? Can we use Optional<TrustManager[]>` instead of a semantic null?

Reply:
The TMF returns an array. I'm not sure what circumstances may it can return >1.  Maybe sticking with an array is safer.
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/javax/net/ssl/TrustManagerFactory.html#getTrustManagers()

Reply:
Ah if it's not our api then leave it as an array. I'm on the fence about null handling. I'd rather get rid of it but equally API consistency is important